src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,117 @@
+import duke.command.MarkCommand;
+import duke.command.UnMarkCommand;
+import duke.common.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+/**
+ * Deals with parsing the user input.
+ *
+ * @author Rama Aryasuta Pangestu
+ */
+public abstract class Parser {
+    /**
+     * Returns a <code>Command</code> equivalent to the user input.
+     *
+     * @param input the user input
+     * @return a <code>Command</code> equivalent to the user input
+     * @throws DukeException if the input format is invalid
+     */
+    public static Command parse(String input) throws DukeException {
+        String[] args = input.trim().split("\\s+");
+        Command command;
+        switch (args[0]) {
+        case "bye":
+            command = new ExitCommand();
+            break;
+        case "list":
+            command = new ListCommand();
+            break;
+        case "find":
+            command = new FindCommand(
+                    Arrays.stream(args).skip(1).reduce("", (x, y) -> x + " " + y).trim());
+            break;
+        case "mark":
+            try {
+                command = new MarkCommand(Integer.parseInt(args[1]) - 1);
+            } catch (IndexOutOfBoundsException exception) {
+                throw new DukeException("OOPS!!! No task index is specified :(");
+            } catch (NumberFormatException exception) {
+                throw new DukeException("OOPS!!! You didn't give a valid index :(");
+            }
+            break;
+        case "unmark":
+            try {
+                command = new UnMarkCommand(Integer.parseInt(args[1]) - 1);
+            } catch (IndexOutOfBoundsException exception) {
+                throw new DukeException("OOPS!!! No task index is specified :(");
+            } catch (NumberFormatException exception) {
+                throw new DukeException("OOPS!!! You didn't give a valid index :(");
+            }
+            break;
+        case "delete":
+            try {
+                command = new DeleteCommand(Integer.parseInt(args[1]) - 1);
+            } catch (IndexOutOfBoundsException exception) {
+                throw new DukeException("OOPS!!! No task index is specified :(");
+            } catch (NumberFormatException exception) {
+                throw new DukeException("OOPS!!! You didn't give a valid index :(");
+            }
+            break;
+        case "todo":
+            command = new AddCommand(new ToDo(
+                    Arrays.stream(args).skip(1).reduce("", (x, y) -> x + " " + y).trim(),
+                    false));
+            break;
+        case "event":
+            try {
+                String description = Arrays.stream(args)
+                        .takeWhile(x -> !x.equals("/at")).skip(1)
+                        .reduce("", (x, y) -> x + " " + y).trim();
+                String[] timeArgument = Arrays.stream(args)
+                        .dropWhile(x -> !x.equals("/at")).toArray(String[]::new);

```

Comment:
Perhaps this few lines can be shortened by abstracting them out into a helper function? Similar statements are found in other switch cases as well.